Others/Dijkstra.java

```diff
@@ -1,192 +1,52 @@
-        if (!graph.containsKey(startName)) {
-            System.err.printf("Graph doesn't contain start vertex \"%s\"\n", startName);
-            return;
-        }
-        final Vertex source = graph.get(startName);
-        NavigableSet<Vertex> q = new TreeSet<>();
-
-        // set-up vertices
-        for (Vertex v : graph.values()) {
-            v.previous = v == source ? source : null;
-            v.dist = v == source ? 0 : Integer.MAX_VALUE;
-            q.add(v);
-        }
-
-        dijkstra(q);
-    }
-
-    /**
-     * Implementation of dijkstra's algorithm using a binary heap.
-     */
-    private void dijkstra(final NavigableSet<Vertex> q) {
-        Vertex u, v;
-        while (!q.isEmpty()) {
-            // vertex with shortest distance (first iteration will return source)
-            u = q.pollFirst();
-            if (u.dist == Integer.MAX_VALUE)
-                break; // we can ignore u (and any other remaining vertices) since they are unreachable
-
-            // look at distances to each neighbour
-            for (Map.Entry<Vertex, Integer> a : u.neighbours.entrySet()) {
-                v = a.getKey(); // the neighbour in this iteration
-
-                final int alternateDist = u.dist + a.getValue();
-                if (alternateDist < v.dist) { // shorter path to neighbour found
-                    q.remove(v);
-                    v.dist = alternateDist;
-                    v.previous = u;
-                    q.add(v);
-                }
-            }
-        }
-    }
-
-    /**
-     * Prints a path from the source to the specified vertex
-     */
-    public void printPath(String endName) {
-        if (!graph.containsKey(endName)) {
-            System.err.printf("Graph doesn't contain end vertex \"%s\"\n", endName);
-            return;
-        }
-
-        graph.get(endName).printPath();
-        System.out.println();
-    }
 
-    /**
-     * Prints the path from the source to every vertex (output order is not guaranteed)
-     */
-    public void printAllPaths() {
-        for (Vertex v : graph.values()) {
-            v.printPath();
-            System.out.println();
-        }
-    }
+	public static void djikstra(int a[][]) {
+		boolean visited[]=new boolean[a.length];
+		int dist[]=new int[a.length];
+		dist[0]=0;
+		for(int i=1;i<a.length;i++)
+			dist[i]=Integer.MAX_VALUE;
+		for(int i=0;i<a.length;i++) {
+			int min=findMinVertex(dist,visited);
+			visited[min]=true;
+			for(int j=0;j<a.length;j++) {
+				if(a[min][j]!=0 && !visited[j]) {
+					int d=dist[min]+a[min][j];
+					if(d<dist[j]) {
+						dist[j]=d;
+					}
+				}
+			}
+		}
+		for(int i=0;i<a.length;i++) {
+			System.out.println(i+" "+dist[i]);
+		}
+	}
+	
+	public static int findMinVertex(int a[],boolean visited[]) {
+		int min=-1;
+		for(int i=0;i<a.length;i++) {
+			if(!visited[i] && (min==-1 || a[i]<a[min]))
+				min=i;
+		}
+		return min;
+	}
+
+	public static void main(String[] args) {

```

Comment:
Try not to use this approach of having 
```java
public static void main(String[] args) {...}
```
call your actual method. 

Instead create your class while keeping in mind of how others will use this i.e. by creating objects and calling your actual methods. So make your methods 
```java
public <return-type> methodName()
```

Please make this changes.